On x86_64, cygwin uses the msabi calling convention, not sysvabi calling convention

--- origsrc/Mesa-8.0.5/src/gallium/auxiliary/rtasm/rtasm_x86sse.h	2011-11-08 18:09:41.000000000 +0000
+++ src/Mesa-8.0.5/src/gallium/auxiliary/rtasm/rtasm_x86sse.h	2013-10-03 17:08:21.274600000 +0100
@@ -139,7 +139,7 @@ static INLINE enum x86_target x86_target
 {
 #ifdef PIPE_ARCH_X86
    return X86_32;
-#elif defined(_WIN64)
+#elif (defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_WINDOWS)) && defined(PIPE_ARCH_X86_64)
    return X86_64_WIN64_ABI;
 #elif defined(PIPE_ARCH_X86_64)
    return X86_64_STD_ABI;
